File: app/src/main/java/com/frama/miserend/hu/di/modules/RouterModule.java
package com.frama.miserend.hu.di.modules;

import android.app.Activity;

import com.frama.miserend.hu.di.scopes.PerActivity;
import com.frama.miserend.hu.router.Router;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Balazs on 2018. 02. 12..
 */
@Module
public class RouterModule {

    @Provides
    @PerActivity
    Router provideRouter(Activity activity) {
        return new Router(activity);
    }
}

File: app/src/main/java/com/frama/miserend/hu/di/modules/RouterModule.kt
package com.frama.miserend.hu.di.modules

import android.app.Activity

import com.frama.miserend.hu.di.scopes.PerActivity
import com.frama.miserend.hu.router.Router

import dagger.Module
import dagger.Provides

/**
 * Created by Balazs on 2018. 02. 12..
 */
@Module
class RouterModule {

    @Provides
    @PerActivity
    internal fun provideRouter(activity: Activity): Router {
        return Router(activity)
    }
}
